src/app/pages/signing/signing.component.ts

```diff
@@ -48,65 +33,34 @@ export class SigningComponent implements OnInit {
       this.uid = params.uid;
       this.key = params.key;
     });
-    this.maxCharCount = MsgIoService.maxMessageLength;
     this.authService.displayName(this.uid).then(userName => this.name = userName).catch(error => alert(error));
+    this.signingService.resetDefaultValues();
   }
 
-  /**
-   * @param font - Font selected from dropdown menu
-   */
-  selectFont(font: string) {
-    this.currentFont = font;
-    this.currentFontFamily = font + ',' + this.fontDict[font].backupFont;
-    this.currentFontDisplay = this.fontDict[font].fontFamily;
-  }
-
-  /**
-   * The following functions add text formatting characters around selected text within textbox.
-   * @param textbox - textbox in which user types.
-   * @param formatCharacters - formatting character(s) to be placed around the selected text
-   * @param endCharacter - Optional - The ending character, if different from the starting character
-   */
-  addFormatting(textbox: HTMLTextAreaElement, formatCharacters: string, endCharacter?: string) {
-    const start = textbox.selectionStart;
-    const end = textbox.selectionEnd;
-    const text = textbox.value;
-
-    // If the end character was not provided, we assume it will be the same as the start character
-    if (typeof endCharacter === 'undefined' || !endCharacter) {
-      endCharacter = formatCharacters;
-    }
-
-    this.signingBoxText = text.slice(0, start) + formatCharacters + text.slice(start, end) + endCharacter +
-      text.slice(end);
-    textbox.select();
-  }
-
-  /**
-   * Update character count value and text colour
-   * @param textbox - textbox in which user types.
-   */
-  updateCount(textbox: HTMLTextAreaElement) {
-    this.charCount = textbox.value.length;
-    this.charCountColor = (this.charCount > this.maxCharCount) ? '#EE1111' : '#b0b0b0';
-    this.canSend = (0 < this.charCount && this.charCount <= this.maxCharCount);
+  // @HostListener allows us to guard against browser refresh, close, etc.
+  @HostListener('window:beforeunload')
+  canDeactivate(): Observable<boolean> | boolean {
```

Comment:
@TommyHuang-dev can you change this to proper TS documentation above the function instead of within?